File: howl-application/src/main/java/io/quartic/howl/HowlConfiguration.java
package io.quartic.howl;

import io.dropwizard.Configuration;
import io.quartic.howl.storage.StorageBackend;

public class HowlConfiguration extends Configuration {
    StorageBackendConfig storageBackendConfig;

    public void setStorage(StorageBackendConfig storageBackendConfig) {
        this.storageBackendConfig = storageBackendConfig;
    }

    public StorageBackendConfig getStorage() {
        return storageBackendConfig;
    }
}

File: howl/src/main/kotlin/io/quartic/howl/HowlConfiguration.kt
package io.quartic.howl

import io.quartic.common.application.ConfigurationBase
import io.quartic.common.secrets.EncryptedSecret
import io.quartic.howl.storage.StorageConfig

data class HowlConfiguration(
    val aws: AwsConfiguration? = null,
    val namespaces: Map<String, StorageConfig> = emptyMap()
) : ConfigurationBase() {
    data class AwsConfiguration(
        val region: String,
        val accessKeyId: String,
        val secretAccessKeyEncrypted: EncryptedSecret
    )
}
